pkg/istio-agent/sds-agent.go

```diff
@@ -427,11 +431,11 @@ func applyEnvVars() {
 	serverOptions.CAEndpoint = caEndpointEnv
 	serverOptions.TrustDomain = trustDomainEnv
 	serverOptions.Pkcs8Keys = pkcs8KeysEnv
+	serverOptions.RecycleInterval = staledConnectionRecycleIntervalEnv
 	workloadSdsCacheOptions.SecretTTL = secretTTLEnv
 	workloadSdsCacheOptions.SecretRefreshGraceDuration = secretRefreshGraceDurationEnv
 	workloadSdsCacheOptions.RotationInterval = secretRotationIntervalEnv
-
-	serverOptions.RecycleInterval = staledConnectionRecycleIntervalEnv
-
 	workloadSdsCacheOptions.InitialBackoffInMilliSec = int64(initialBackoffInMilliSecEnv)
+	// Disable the secret eviction for istio agent.
+	workloadSdsCacheOptions.EvictionDuration = 3650 * 24 * time.Hour
```

Comment:
Should we just add a check for `if EvictionDuration != 0 { do eviction...}`?

Reply:
I can do that.